src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskListTabPanel.tsx

```diff
@@ -3,28 +3,201 @@
  * See the LICENSE file in the repository root folder for details.
  */
 
-import { Box, Link, Typography } from '@mui/material';
+import { Box, IconButton, Link, ListItemIcon, Menu, MenuItem, Typography } from '@mui/material';
+import { DataGrid, GridColDef, GridColumnHeaderParams } from '@mui/x-data-grid';
+import { RoleEnum, Task, TaskStatus, UserPreview } from 'shared';
+import { datePipe } from '../../../utils/pipes';
+import { GridColDefStyle } from '../../../utils/tables';
+import MenuIcon from '@mui/icons-material/Menu';
+import { useState } from 'react';
+import PauseIcon from '@mui/icons-material/Pause';
+import DeleteIcon from '@mui/icons-material/Delete';
+import PlayArrowIcon from '@mui/icons-material/PlayArrow';
+import CheckIcon from '@mui/icons-material/Check';
+import { useAuth } from '../../../hooks/auth.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
 
 interface TaskListTabPanelProps {
   index: number;
   value: number;
+  tasks: Task[];
+  status: TaskStatus;
 }
 
 const TaskListTabPanel = (props: TaskListTabPanelProps) => {
-  const { value, index } = props;
+  const { value, index, tasks, status } = props;
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const auth = useAuth();
+
+  if (!auth.user) return <LoadingIndicator />;
+
+  const disabled = auth.user.role === RoleEnum.GUEST;
+
+  const handleMenu = (event: React.MouseEvent<HTMLElement>) => setAnchorEl(event.currentTarget);
+  const handleClose = () => setAnchorEl(null);
 
   // Skeleton copied from https://mui.com/material-ui/react-tabs/.
   // If they release the TabPanel component from @mui/lab to @mui/material then change the div to TabPanel.
 
+  const namePipe = (user: UserPreview) => {
+    return user.firstName + ' ' + user.lastName + ', ';
+  };
+
+  const renderActionButtons = () => (
+    <>
+      <IconButton onClick={handleMenu}>
+        <MenuIcon />
+      </IconButton>
+      <Menu
+        id="task-appbar"
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'center',
+          horizontal: 'center'
+        }}
+        PaperProps={{
+          style: {
+            transform: 'translateX(-10%) translateY(35%)'
+          }
+        }}
+        transformOrigin={{
+          vertical: 'center',
+          horizontal: 'center'
+        }}
+        sx={{ minHeight: 0 }}
+        open={Boolean(anchorEl)}
+        onClose={handleClose}
+      >
+        <MenuItem disabled>
+          <ListItemIcon>
+            <DeleteIcon fontSize="small" />
+          </ListItemIcon>
+          Delete
+        </MenuItem>
+        {status !== TaskStatus.IN_BACKLOG && (
```

Comment:
this is actually only if the task is in progress. If its done its not allowed to be moved to backlog, supposed to be like one jump at a time.